core/internal/stream/sender.go

```diff
@@ -269,6 +269,7 @@ func NewSender(
 			params.Logger,
 			params.GraphqlClient,
 			params.FileTransferManager,
+			params.FeatureProvider,
```

Comment:
why do you need the entire feature provider? you could just pass the relevant flags i think